docs/providers.md

```diff
@@ -0,0 +1,114 @@
+---
+title: Providers
+toc: true
+weight: 400
+indent: true
+---
+
+# Providers
+
+Providers are external infrastructure providers such as Amazon Web Services,
+Microsoft Azure, Google Cloud and Alibaba Cloud.
+
+In order to provision a resource in the provider, a Custom Resource Definition(CRD)
+needs to be registered in your Kubernetes cluster and its controller should
+be watching instances of that Custom Resource Definition. Provider packages
+contain many Custom Resource Definitions and their controllers to extend
+functionality of Crossplane.
+
+Here is the list of current providers:
+
+* [Amazon Web Services][provider-aws]
+  * [API Reference][aws-reference]
+* [Google Cloud][provider-gcp]
+  * [API Reference][gcp-reference]
+* [Microsoft Azure][provider-azure]
+  * [API Reference][azure-reference]
+* [Rook][provider-rook]
+  * [API Reference][rook-reference]
+* [Alibaba Cloud][provider-alibaba]
+  * [API Reference][alibaba-reference]
+
+## Installation
+
+Since all managed resources are all cluster-scoped, provider packages are
+bundled as `ClusterPackage`s. In order to request installation of a provider,
+the following `ClusterPackageInstall` needs to be created:
+
+```yaml
+apiVersion: packages.crossplane.io/v1alpha1
+kind: ClusterPackageInstall
+metadata:
+  name: provider-gcp
+  namespace: crossplane-system
+spec:
+  package: "crossplane/provider-gcp:master"
+```
+
+The field `spec.package` is where you refer to the container image of the
+provider. Crossplane Package Manager will unpack that container, register
+CRDs and set up necessary RBAC rules and then start the controllers.
+
+There are a few ways that are more convenient than creating a YAML to install
+the providers.
```

Comment:
You imply that the `ClusterPackageInstall` is less convenient but it's not clear why. I'd consider structuring this whole section like "You can install a Crossplane infrastructure provider via helm, the crossplane kubectl plugin, or by apply a Crossplane `ClusterPackageInstall` resource." Then you can just bullet point these 3 options out and people can check out the option that sounds like it fits their style. Do we have a preferred means of installing?

Reply:
I suggest we pick our preferred method here and not worry too much about documenting the alternatives. To my knowledge submitting a `ClusterPackageInstall` is our preferred method. I'd slightly prefer recommending the `crossplane package install` command, but that would require the user to have `crossplane-cli` installed.

Reply:
If we can do tabs or similar would be nice to document the 3 install options, so new users are exposed to the crossplane-cli while also understanding that everything can be done with CRs against the Kubernetes API, and that we also offer the convenience of installing optional cluster-scoped packages when installing the Crossplane helm chart.

Reply:
Supporting tabs in the markdown appears to be a little more difficult given our jekyll setup than I originally expected. However, if we get the content to a good place then I could circle back around on Monday and try and move some of the stuff to tabs. For the time-being I have just been doing separate sections but with links to jump to each. I also want to check how jekyll handles `<summary>` in markdown :+1: